packages/firebase_messaging/example/test_driver/firebase_messaging.dart

```diff
@@ -11,6 +18,60 @@ void main() {
   group('$FirebaseMessaging', () {
     final FirebaseMessaging firebaseMessaging = FirebaseMessaging();
 
+    // Uncomment this when testing onMessage callback.
+//    setUpAll(() async {
+//      await firebaseMessaging.requestNotificationPermissions(
+//        const IosNotificationSettings(sound: true, badge: true, alert: true),
+//      );
+//    });
+
+    // We skip this test because it requires a valid server token specified at
+    // the top of this file uncommenting `setUpAll`, and requires agreeing to
+    // receive messages by hand on ios.
+    test('onMessage', () async {
+      await http.post(
+        'https://fcm.googleapis.com/fcm/send',
+        headers: <String, String>{
+          'Content-Type': 'application/json',
+          'Authorization': 'key=$serverToken',
+        },
+        body: jsonEncode(
+          <String, dynamic>{
+            'notification': <String, dynamic>{
+              'body': 'this is a body',
+              'title': 'this is a title'
+            },
+            'priority': 'high',
+            'data': <String, dynamic>{
+              'click_action': 'FLUTTER_NOTIFICATION_CLICK',
+              'id': '1',
+              'status': 'done'
+            },
+            'to': await firebaseMessaging.getToken(),
+          },
+        ),
+      );
+
+      final Completer<Map<String, dynamic>> completer =
+          Completer<Map<String, dynamic>>();
+
+      firebaseMessaging.configure(
+        onMessage: (Map<String, dynamic> message) async {
+          completer.complete(message);
+        },
+      );
+
+      final Map<String, dynamic> message = await completer.future;
+      expect(
+        message,
+        containsPair('notification', containsPair('title', 'this is a title')),
+      );
+      expect(
+        message,
+        containsPair('notification', containsPair('body', 'this is a body')),
+      );
+    }, timeout: const Timeout(Duration(seconds: 5)), skip: true);
```

Comment:
Since this is skipped for the reasons you identified, could we remove it till we have a good way to run it. Maybe even include it in the README as an option test for developers?